### ISSUE TITLE
ScalaTest + Scalactic 3.0 upgrade

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#269]
already mentioned on https://github.com/scala/scala-dev/issues/203

**Comment by SethTisue** [comment]
yuck, the 3.0.x branch won't even extract; dbuild finds a cycle in its subproject graph

hypothesis: could be because dbuild isn't taking `% "test"` into account. (ScalaTest depends on Scalactic; and Scalactic itself has tests that use ScalaTest; but it isn't a cycle as long as the non-test projects are built before the test ones.) I don't have any hard evidence for this, it's just a hunch. the cycles in the subproject graph dbuild dumps out seem to go through test projects

hmm, or, poring over their (lengthy!) build, the way `aggregate` was used seemed suspicious to me. maybe https://github.com/SethTisue/scalatest/commit/f620c113b58efbf390d7601ac4ace9af3dd63662 will fix it? not sure what the `aggregate` calls are intended to accomplish, but perhaps they're introducing a dependency of non-test on test stuff, causing the cycle

**Comment by SethTisue** [comment]
taking the `aggregate` calls out makes extraction succeed:

```
[scalatest] [warn] Originally requested: scalatest, scalactic
[scalatest] [warn] Now added: scalacticMacro
```

but https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/516/console shows:

```
[scalatest] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/scalatest-e9bb9cf9971cb2b4809e4a367dbcf6bd3471ee38/scalatest/target/docsrc/src/main/scala/org/scalatest/selenium/WebBrowser.scala:4610: recursive value firefoxProfile needs type
[scalatest] [error]   implicit val webDriver = new FirefoxDriver(firefoxProfile)
[scalatest] [error]                                              ^
[scalatest] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/scalatest-e9bb9cf9971cb2b4809e4a367dbcf6bd3471ee38/scalatest/target/docsrc/src/main/scala/org/scalatest/selenium/WebBrowser.scala:4616: could not find implicit value for parameter driver: org.openqa.selenium.WebDriver
[scalatest] [error]     capture to directory
[scalatest] [error]             ^
```

looks fixable, but I think I should set this aside until after 2.12.0-RC1 is out the door

oh wait, it's fixed in Adriaan's fork, adriaanm/scalatest@3668daf1144a7e078585c1e1d9145b55e5200092

**Comment by SethTisue** [comment]
ok that worked, just for ScalaTest & Scalactic, so here's a full run, surely the upgrade will break things... https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/518/consoleFull

**Comment by SethTisue** [comment]
oh and hey, pleasant surprise, this came really close to passing, just one failure in scalamock:

```
[scalamock] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/scalamock-8d8139cab9145d0a38d9f141f750f03c3fbcbd0a/frameworks/scalatest/src/main/scala/org/scalamock/scalatest/AbstractMockFactory.scala:31: not found: type NoArgTest
[scalamock] [error]   abstract override def withFixture(test: NoArgTest): Outcome = {
[scalamock] [error]                                           ^
[scalamock] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/scalamock-8d8139cab9145d0a38d9f141f750f03c3fbcbd0a/frameworks/scalatest/src/main/scala/org/scalamock/scalatest/AbstractMockFactory.scala:31: not found: type NoArgTest
[scalamock] [error]   abstract override def withFixture(test: NoArgTest): Outcome = {
[scalamock] [error]                                           ^
[scalamock] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/scalamock-8d8139cab9145d0a38d9f141f750f03c3fbcbd0a/frameworks/scalatest/src/main/scala/org/scalamock/scalatest/AbstractMockFactory.scala:35: value withFixture is not a member of org.scalatest.SuiteMixin with org.scalamock.MockFactoryBase
[scalamock] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/scalamock-8d8139cab9145d0a38d9f141f750f03c3fbcbd0a/frameworks/scalatest/src/main/scala/org/scalamock/scalatest/AbstractMockFactory.scala:35: value withFixture is not a member of org.scalatest.SuiteMixin with org.scalamock.MockFactoryBase
[scalamock] [error]         val outcome = super.withFixture(test)
[scalamock] [error]         val outcome = super.withFixture(test)
[scalamock] [error]                             ^
[scalamock] [error]                             ^
[scalamock] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/scalamock-8d8139cab9145d0a38d9f141f750f03c3fbcbd0a/frameworks/scalatest/src/main/scala/org/scalamock/scalatest/AbstractMockFactory.scala:46: value withFixture is not a member of org.scalatest.SuiteMixin with org.scalamock.MockFactoryBase
[scalamock] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/scalamock-8d8139cab9145d0a38d9f141f750f03c3fbcbd0a/frameworks/scalatest/src/main/scala/org/scalamock/scalatest/AbstractMockFactory.scala:46: value withFixture is not a member of org.scalatest.SuiteMixin with org.scalamock.MockFactoryBase
[scalamock] [error]       super.withFixture(test)
[scalamock] [error]       super.withFixture(test)
[scalamock] [error]             ^
[scalamock] [error]             ^
[scalamock] [error] three errors found
```

**Comment by SethTisue** [comment]
fixed by #271
